packages/@aws-cdk/aws-lambda-nodejs/lib/bundling.ts

```diff
@@ -57,11 +59,13 @@ export class Bundling implements cdk.BundlingOptions {
   private readonly relativeEntryPath: string;
   private readonly relativeTsconfigPath?: string;
   private readonly externals: string[];
+  private readonly packageManager: PackageManager;
 
   constructor(private readonly props: BundlingProps) {
-    Bundling.runsLocally = Bundling.runsLocally
-      ?? getEsBuildVersion()?.startsWith(ESBUILD_VERSION)
-      ?? false;
+    this.packageManager = PackageManager.fromLockFile(props.depsLockFilePath);
+
+    Bundling.esbuildInstallation = Bundling.esbuildInstallation ?? EsbuildInstallation.detect();
+    const runsLocally = !!Bundling.esbuildInstallation?.version.startsWith(ESBUILD_VERSION);
```

Comment:
maybe `esbuildInstallation?.isLocal`

Reply:
There too many locals... this `runsLocally` here means local bundling (vs Docker bundling). The `isLocal` of the `esbuildInstallation` means is locally installed (vs globally installed).